core/src/avm2/object/xml_list_object.rs

```diff
@@ -67,62 +71,62 @@ impl<'gc> XmlListObject<'gc> {
         target_object: Option<XmlOrXmlListObject<'gc>>,
         target_property: Option<Multiname<'gc>>,
     ) -> XmlListObject<'gc> {
-        let base = ScriptObjectData::new(activation.context.avm2.classes().xml_list);
-        XmlListObject(GcCell::new(
+        let base = ScriptObjectData::new(activation.context.avm2.classes().xml_list).into();
+        XmlListObject(Gc::new(
             activation.context.gc_context,
             XmlListObjectData {
                 base,
-                children,
-                target_object,
-                target_property,
-                target_dirty: false,
+                children: RefLock::new(children),
+                target_object: Lock::new(target_object),
+                target_property: RefLock::new(target_property),
+                target_dirty: Cell::new(false),
             },
         ))
     }
 
-    pub fn set_dirty_flag(&self, mc: &Mutation<'gc>) {
-        self.0.write(mc).target_dirty = true;
+    pub fn set_dirty_flag(&self) {
+        self.0.target_dirty.set(true);
     }
 
     pub fn length(&self) -> usize {
-        self.0.read().children.len()
+        self.0.children.borrow().len()
     }
 
     pub fn xml_object_child(
         &self,
         index: usize,
         activation: &mut Activation<'_, 'gc>,
     ) -> Option<XmlObject<'gc>> {
-        let mut write = self.0.write(activation.context.gc_context);
-        if let Some(child) = write.children.get_mut(index) {
+        let mut children = self.children_mut(activation.gc());
+        if let Some(child) = children.get_mut(index) {
             Some(child.get_or_create_xml(activation))
         } else {
             None
         }
     }
 
     pub fn node_child(&self, index: usize) -> Option<E4XNode<'gc>> {
-        self.0.read().children.get(index).map(|x| x.node())
+        self.0.children.borrow().get(index).map(|x| x.node())
     }
 
     pub fn children(&self) -> Ref<'_, Vec<E4XOrXml<'gc>>> {
-        Ref::map(self.0.read(), |d| &d.children)
+        self.0.children.borrow()
     }
 
     pub fn children_mut(&self, mc: &Mutation<'gc>) -> RefMut<'_, Vec<E4XOrXml<'gc>>> {
-        RefMut::map(self.0.write(mc), |d| &mut d.children)
+        unlock!(Gc::write(mc, self.0), XmlListObjectData, children).borrow_mut()
     }
 
     pub fn set_children(&self, mc: &Mutation<'gc>, children: Vec<E4XOrXml<'gc>>) {
-        self.0.write(mc).children = children;
+        *unlock!(Gc::write(mc, self.0), XmlListObjectData, children).borrow_mut() = children;
     }
 
     pub fn target_object(&self) -> Option<XmlOrXmlListObject<'gc>> {
-        self.0.read().target_object
+        self.0.target_object.get()
     }
 
     pub fn target_property(&self) -> Option<Multiname<'gc>> {
-        self.0.read().target_property.clone()
+        self.0.target_property.borrow().clone()
```

Comment:
I think `target_property` can be a `Lock` too, if you `clone` it here?

Reply:
Can't clone because `Cell<T>` only implements Clone for T: Copy